src/ResourceManager/Network/Commands.Network/ServiceEndpointPolicy/NewAzureServiceEndpointPolicyCommand.cs

```diff
@@ -0,0 +1,98 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.Network
+{
+    using System.Collections.Generic;
+    using System.Management.Automation;
+    using Microsoft.Azure.Commands.Network.Models;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using MNM = Microsoft.Azure.Management.Network.Models;
+    using Management.Network;
+
+    [Cmdlet(VerbsCommon.New, "AzureRmServiceEndpointPolicy", SupportsShouldProcess = true), OutputType(typeof(PSServiceEndpointPolicy))]
+    public class NewAzureServiceEndpointPolicyCommand : ServiceEndpointPolicyBaseCmdlet
+    {
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The name of the subnet")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "List of service endpoint definitions")]
+        [ValidateNotNullOrEmpty]
+        public List<PSServiceEndpointPolicyDefinition> ServiceEndpointPolicyDefinition { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource group name.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public virtual string ResourceGroupName { get; set; }
+
+        [Parameter(
+         Mandatory = true,
+         ValueFromPipelineByPropertyName = true,
+         HelpMessage = "location.")]
+        [ValidateNotNullOrEmpty]
+        public virtual string Location { get; set; }
```

Comment:
use the [LocationCompleter](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/azure-powershell-developer-guide.md#location-completer)